business/apps.go

```diff
@@ -48,13 +48,160 @@ func buildFinalLabels(m map[string][]string) map[string]string {
 }
 
 // GetAppList is the API handler to fetch the list of applications in a given namespace
-func (in *AppService) GetAppList(namespace string) (models.AppList, error) {
+func (in *AppService) GetAppList(namespace string, linkIstioResources bool) (models.AppList, error) {
 	appList := &models.AppList{
 		Namespace: models.Namespace{Name: namespace},
 		Apps:      []models.AppListItem{},
 	}
-	apps, err := fetchNamespaceApps(in.businessLayer, namespace, "")
-	if err != nil {
+
+	var err error
+	var apps namespaceApps
+
+	var virtualServices []kubernetes.IstioObject
+	var destinationRules []kubernetes.IstioObject
+	var gateways []kubernetes.IstioObject
+	var authorizationPolicies []kubernetes.IstioObject
+	var peerAuthentications []kubernetes.IstioObject
+	var sidecars []kubernetes.IstioObject
+	var requestAuthentications []kubernetes.IstioObject
+	var envoyFilters []kubernetes.IstioObject
+
+	nFetches := 1
+	if linkIstioResources {
+		nFetches = 9
+	}
+
+	wg := sync.WaitGroup{}
+	wg.Add(nFetches)
+	errChan := make(chan error, nFetches)
+
+	go func() {
+		defer wg.Done()
+		var err2 error
+		apps, err2 = fetchNamespaceApps(in.businessLayer, namespace, "")
+		if err2 != nil {
+			log.Errorf("Error fetching Applications per namespace %s: %s", namespace, err2)
+			errChan <- err2
+		}
+	}()
+
+	if linkIstioResources {
+		go func() {
+			defer wg.Done()
+			var err2 error
+			if IsNamespaceCached(namespace) {
+				virtualServices, err2 = kialiCache.GetIstioObjects(namespace, kubernetes.VirtualServices, "")
+			} else {
+				virtualServices, err2 = in.k8s.GetIstioObjects(namespace, kubernetes.VirtualServices, "")
+			}
+			if err2 != nil {
+				log.Errorf("Error fetching Istio VirtualServices per namespace %s: %s", namespace, err)
```

Comment:
Shouldn't this log print `err2` instead? If so, it should be amended in other similar LOCs.